main.pm

```diff
@@ -83,8 +83,8 @@ sub cleanup_needles() {
     remove_desktop_needles("minimalx");
     remove_desktop_needles("textmode");
 
-    remove_flavor_needles('Server-DVD');
-    remove_flavor_needles('Desktop-DVD');
+    remove_flavor_needles('Server-DVD') if ( !check_var( 'FLAVOR', 'Server-MINI' ) );
+    remove_flavor_needles('Desktop-DVD') if ( !check_var( 'FLAVOR', 'Desktop-MINI') );
     remove_flavor_needles('Core-DVD');
```

Comment:
FLAVOR=Server-MINI and FLAVOR=Desktop-MINI are different flavors, so I think we should have two new lines here

remove_flavor_needles('Server-MINI')
remove_flavor_needles('Desktop-MINI')

I don't think this "if ( !check_var ..." approach on lines 86 and 87 are the correct way.